dtfs-central-api/src/services/tfm/deal-cancellation.service.ts

```diff
@@ -70,7 +80,13 @@ export class DealCancellationService {
       dealType,
     });
 
-    return response;
+    await Promise.all(
+      riskExpiredFacilities.map(({ _id: facilityId }) =>
+        PortalFacilityService.updateStatus({ facilityId, status: FACILITY_STATUS.RISK_EXPIRED, dealType, auditDetails }),
```

Comment:
I haven't worked with Mongo for a while. Are we able to execute a bulk operation instead of mapping for individual calls?

Reply:
Its possible using a `updateMany` call but I cant find any existing functions doing this. I'd like to keep the scope of this smaller & reuse existing functions, rather than creating a new `updateFacilitiesByDealId` function. Whats your thoughts?

Reply:
Thinking about it, I definitely want to call the existing update BSS/EWCS facility function, since it adds a few extra fields & Id want to keep that logic DRY

Reply:
Do you know what the typical average and maximum amount of facilities we've seen?

Reply:
Asking on slack :)

Reply:
Abhi reckons 10 is about the most that is practical

Reply:
Raised a subtask for this https://ukef-dtfs.atlassian.net/browse/DTFS2-7625